en/05_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -136,6 +136,133 @@ class MyClass extends Class
+- Use `SCREAMING_SNAKE_CASE` for constants including class contants, and environment variables
+- Note that there will be some inconsistencies with legacy code where static methods have been a combination of `MyClass::camelCase()` and `MyClass::snake_case()`
+
+### Accessors
+
+- For accessors omit the leading 'get' with getters and prefix with 'set' for setters. e.g. `function myValue():sting { return $this->value; }` and `function setMyValue(string $value): void { $this->value = $value; }`.
+- Note that there will be some inconsistencies with legacy code as get accessors have been a combination of "with get", "without get", and "without get with the first letter uppercased"
+
+### File suffixes and prefixes
+
+- This is for new classes and interfaces that are created. Legacy Silvesrtripe code is inconsistent whether or not it follows these conventions.
+- Suffix the classnames with the type of class that is subclassed. The suffixed is sometimes a shortened version of the parent class because it reads better while retaining easy comprehension. Here are some common examples with the parents classes in brackets
+  - `Exception` (Exception)
+  - `Controller` (Controller)
+  - `Extension` (Extension, DataExtension)
+  - `Admin` (ModelAdmin)
+  - `Handler` (RequestHandler)
+  - `Filter` (SearchFilter)
+  - `Form` (Form)
+  - `Field` (FormField)
+  - `Job` (QueuedJob)
+  - `Task` (BuildTask)
+  - `Service` (generic class type that has no parent class)
+Some parent classes have unorthodox naming, in which case suffix with something more standard
+  - `Page` (SiteTree)
+  - `Controller` (LeftAndMain, CMSMain)
+- Do not suffix classes if they are a subclass any of the following:
+  - `DataObject`
+- Suffix interfaces with `Interface`
+- Suffix traits with `Trait`
+- Prefix abstract classes with `Abstract`
+
+### Other naming conventions
+
+- Extensions should match the name of the class they are extending. For example, an extension class that's added to LeftAndMain should be called LeftAndMainExtension
+- When choosing names for variables and methods use the guiding principle "make it as obvious as possible"
+- Ensure that variable names and method naming is consistent. For example, if you are creating a new class and have a variable named `$itemValue`` which is passed to another method in the same class, the paramater name for that method should also be `$itemValue``.
+
+## Directory naming - src dir
+
+- Put similar types of classes in the same directory. For example, put Controllers in a Controllers directory
+- Put DataObjects in a directory called Models
+- Do not group different class types together in a directory relating to a single feature
+- Use initial caps pluralised for the directory naming e.g. use `Controllers` rather than `Controller` or `controllers`
+
+## Properties and accessors
+
+- Do not use public properites, instead use a combination of private properites and public getters and setters
```

Comment:
This note is more about explicitly _not_ using public properties. I'd say if there's a public getter and setter method, there's no need for a property to ever be protected, since the subclasses can still access and manipulate the value of the property through those public methods. In my view `protected` is only really sensible for methods, not properties.

Happy to hear alternative views on that though.

Reply:
Yes contextual, protected is correct if it makes sense at the time, though here I'm explicitly saying default to private.

Historically it was protected.  Since I've been in the team for the last few years at least it's been private, so I'm making this explicit here

Reply:
I just think you should change `private` to `non-public` or something. It's entirely reasonable that you have a protected property, that may have a public getter that doesn't directly reference the a property, and an extending class wants access to the protected property. I don't think having this document to point back at and say "oh but this says we should make properties private" is good.

Reply:
Saying non-public here is fine

Reply:
Have updated